tinygrad/gradcheck.py

```diff
@@ -0,0 +1,75 @@
+import numpy as np
+from tinygrad.tensor import Tensor
+
+def mask_like(like, mask_inx, mask_value = 1.0):
+  mask = np.zeros_like(like).reshape(-1)
+  mask[mask_inx] = mask_value
+  return mask.reshape(like.shape)
+
+def numerical_jacobian(model, input, eps = 1e-6):
+  """
+  Compute the Jacobian through Finite-Difference Approximation.
+  Somewhat inspired by [1] but not followed closely.
+
+      model : A tinygrad model
+      input : An input
+      eps : Perturbation step
+
+  returns:
+
+      NJ : an approx. of the Jacobian
+
+  [1]: https://timvieira.github.io/blog/post/2017/04/21/how-to-test-gradient-implementations/
+  """
+  output = model.forward(input)
```

Comment:
Why is a model being passed in here? gradcheck shouldn't operate at this level of abstraction, if you need to run it multiple times, perhaps pass in a function instead?